code/game/machinery/computer/communications.dm

```diff
@@ -230,6 +230,19 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 			src.emagged = 0
 			src.updateDialog()
 
+		if("nukerequest") //When there's no other way
+			if(src.authenticated==2)
+				if(CM.cooldownLeft())
+					usr << "Arrays recycling. Please stand by."
+					return
+				var/input = stripped_input(usr, "Please enter the reason for requesting the nuclear self-destruct codes. Misuse of the nuclear request system will not be tolerated under any circumstances.  Transmission does not guarantee a response.", "Self Destruct Code Request.","")
+				if(!input || !(usr in view(1,src)))
+					return
+				Nuke_request(input, usr)
+				usr << "Request sent."
+				log_say("[key_name(usr)] has requested the nuclear codes from Centcomm")
+				priority_announce("The codes for the on-station nuclear self-destruct have been requested by [usr]. Confirmation or denial of this request will be sent shortly.", "Nuclear Self Destruct Codes Requested",'sound/AI/commandreport.ogg')

```

Comment:
You need to add `CM.lastTimeUsed = world.time` after that for the antispam cooldown thing to work.